### PR TITLE
Declare amount private - use magic method to provide access to the old value

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Declare amount private - use magic method to provide access to the old value

Before
----------------------------------------
`_amount` is undeclared & generally unhelpful & a candidate for removal

After
----------------------------------------
It is declared - but private - extrernal get access will not change in the short term - even as we stop using it entirely - due to the use of a magic method & a function that is only used to support that method

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
I do not know if these are affected but these might need to be reviewed

```
ext/biz.jmaconsulting.grantapplications/CRM/Grant/Form/GrantBase.php:    $this->_amount = $this->get('amount');
ext/btcpay/btcpay.php:          'total_amount' => $form->_amount,
ext/uk.co.vedaconsulting.offlinemembershipwizard/CRM/Offlinemembershipwizard/Form/Wizard/Confirm.php:    $this->_amount = $contribution['amount'];
ext/uk.co.vedaconsulting.offlinemembershipwizard/CRM/Offlinemembershipwizard/Form/Wizard/Confirm.php:      if ($this->_amount > 0.0) {
ext/org.civicrm.webtracking/CRM/WebTracking/Utils.php:      'totalAmount' => $form->_totalAmount ?? $form->_amount ?? NULL,
ext/bitpay/bitpay.php:        'total_amount' => $form->_amount,
```

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I don't think they are accessing the event form from when I looked - but if they are the magic method should still `get` what they want for them - ie __get kicks in if you try to access a private property where you otherwise can't

File: CRM/Event/Form/Registration/Confirm.php [hunk 1]
```diff
@@ -45,6 +45,29 @@ class CRM_Event_Form_Registration_Confirm extends CRM_Event_Form_Registration {
    */
   protected $submittableMoneyFields = ['total_amount', 'net_amount', 'non_deductible_amount', 'fee_amount', 'tax_amount', 'amount'];
 
+  /**
+   * @var int|mixed
```
Comment (braders) [inline CRM/Event/Form/Registration/Confirm.php:49]: @eileenmcnaughton We may _think_ this should be an `int` (or a `float`?), but the fact is that for the most part it's not. So i think this should be typed as `array`, or simply `mixed` alone.

File: CRM/Event/Form/Registration/Confirm.php [hunk 2]
```diff
@@ -45,6 +45,29 @@ class CRM_Event_Form_Registration_Confirm extends CRM_Event_Form_Registration {
...
+   */
+  private $_amount;
+
+  /**
+   * Provide support for extensions that are used to being able to retrieve _lineItem
+   *
+   * Note extension should call getPriceSetID() and getLineItems() directly.
```
Comment (braders) [inline CRM/Event/Form/Registration/Confirm.php:56]: @eileenmcnaughton This comment seems like a copy-paste from elsewhere.

File: CRM/Event/Form/Registration/Confirm.php [hunk 3]
```diff
@@ -1325,4 +1348,51 @@ private function calculateAmounts(): array {
...
   }
 
+  /**
+   * Interim refactoring extraction.
+   *
+   * @internal
+   * @return array
+   */
+  private function calculateLegacyAmountArray(): array {
+    CRM_Core_Error::deprecatedWarning('Do not access _amount as it will be removed');
```
Comment (braders) [inline CRM/Event/Form/Registration/Confirm.php:1358]: @eileenmcnaughton I'm not sure we need the `deprecatedWarning` here, as one will already be called from the `__get()`, and `calculateLegacyAmountArray` is private so cannot be called directly.

File: CRM/Event/Form/Registration/Confirm.php (hunk 2 above)
Comment (eileenmcnaughton) [reply CRM/Event/Form/Registration/Confirm.php:56]: @braders snapped

File: CRM/Event/Form/Registration/Confirm.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Event/Form/Registration/Confirm.php:49]: I'll just not type it I think

**Comment by colemanw** [comment]
Ok let's merge this. It's been reviewed, comments have been addressed, and it provides a compatibility layer that's better in than out.

**Pull request merged**